mutation/range_tombstone_list.hh

```diff
@@ -80,6 +78,15 @@ private:
     }
 };
 
+template <>
+struct fmt::formatter<range_tombstone_entry> {
+    constexpr auto parse(format_parse_context& ctx) { return ctx.begin(); }
```

Comment:
What does returning `ctx.begin()` mean?

Reply:
FWIW, personally, i'd lean toward inheriting from `fmt::formatter<std::string_view>`.

Reply:
I think inheriting from `fmt::formatter<std::string_view>` was fine. We can have the best of both world, by creating a formatter base class called `basic_formatter_base` which does what `fmt::formatter<std::string_view>` does today: provide a no-op `parse()` method for convenience.

Reply:
i am not in favor of introducing yet another layer of indirection. switching back to the `fmt::formatter<std::string_view>` base class.

Reply:
> We can have the best of both world, by creating a formatter base class called basic_formatter_base

That's what I was thinking as well. It makes the intent clear.

> i am not in favor of introducing yet another layer of indirection

What's the problem with it?

Reply:
> > i am not in favor of introducing yet another layer of indirection
> 
> What's the problem with it?

IMHO, the helper class is way too thin and it does not deserve a dedicated class. and, i think, instead of staying away from fmtlib by wrapping it with a wrapper, developers probably just need to understand it better just like how we learn and use boost and the standard library. as fmtlib is already being [standardized](https://en.cppreference.com/w/cpp/utility/format/format). it's just the standard library implementations are try to catch up with the related proposals.

a practical reason is, then we will need to include one more header in each source file where we want to use `basic_formatter_base`.

Reply:
>  just need to understand it better

You can justify every complexity by saying that, it's a matter of what we should value more. Cognitive load on developers, or avoiding thin wrappers?


> a practical reason is, then we will need to include one more header in each source file where we want to use basic_formatter_base.

We could include that header instead of, rather than in addition to.